dali/operators/generic/reshape.cc

```diff
@@ -63,7 +63,20 @@ If a value is not specified, if number of dimension matches existing layout, the
 layout is preserved. If the number of dimensions does not match, the argument is reset
 to empty. If a value is set, and is not empty, the layout must match the dimensionality
 of the output.)code",
-                  TensorLayout(""));
+                  TensorLayout(""))
+  .AddOptionalArg("src_dims", R"code(Indices of dimensions to keep.
+
+This argument can be used to manipulate the order of existing dimensions or to remove or
+add dimensions with extent one. A special index value -1 can be used to insert new dimensions
+at any particular position.
+
+For example, reshaping a sample with shape ``[300, 200, 1]`` and a ``src_dims``
+argument ``[-1, 1, 0]`` produces an output shape ``[1, 200, 300]``. A leading dimension with
+extent 1 is inserted at the beginning, followed by the first original dimensions but in reverse
+order. The last dimension is removed.
+
```

Comment:
I think we were to support reindexing with `rel_shape`, too? I think it deserves a mention.
```
The `src_dims` argument can be used together with `rel_shape`, in which case the relative extents in `rel_shape` describe to the target dimensions. In the example above, specifying ``rel_shape = [-1, 0.5, 2]`` would result in the output shape ``[1, 100, 600]``.
```